domain/src/test/java/org/oppia/android/domain/spotlight/SpotlightStateControllerTest.kt

```diff
@@ -93,37 +96,47 @@ class SpotlightStateControllerTest {
   @Test
   fun testMarkSpotlightState_validFeature_notYetMarked_returnsSuccess() {
     val markSpotlightProvider =
-      spotlightStateController.markSpotlightViewed(profileId, FIRST_CHAPTER)
-    dataProviderTestMonitor.waitForNextSuccessfulResult(markSpotlightProvider)
+      spotlightStateController.markSpotlightViewed(profileId0, FIRST_CHAPTER)
+    val monitor = dataProviderTestMonitor.createMonitor(markSpotlightProvider)
+    testCoroutineDispatchers.runCurrent()
+    monitor.ensureNextResultIsSuccess()
   }
 
   @Test
   fun testMarkSpotlightState_validFeature_alreadyMarked_returnsSuccess() {
     markSpotlightSeen(FIRST_CHAPTER)
     val markSpotlightProvider =
-      spotlightStateController.markSpotlightViewed(profileId, FIRST_CHAPTER)
-    dataProviderTestMonitor.waitForNextSuccessfulResult(markSpotlightProvider)
+      spotlightStateController.markSpotlightViewed(profileId0, FIRST_CHAPTER)
+    val monitor = dataProviderTestMonitor.createMonitor(markSpotlightProvider)
+    testCoroutineDispatchers.runCurrent()
+    monitor.ensureNextResultIsSuccess()
   }
 
   @Test
   fun testMarkSpotlightView_invalidFeature_returnsFailure() {
     val invalidFeature = Spotlight.FeatureCase.FEATURE_NOT_SET
     val markSpotlightProvider =
-      spotlightStateController.markSpotlightViewed(profileId, invalidFeature)
-    dataProviderTestMonitor.waitForNextFailureResult(markSpotlightProvider)
+      spotlightStateController.markSpotlightViewed(profileId0, invalidFeature)
+    assertThrows(SpotlightStateController.SpotlightFeatureNotFoundException::class) {
```

Comment:
This doesn't look right--we should verify the except returned by ``ensureNextResultIsFailing`` is the correct exception with the correct cause message. There shouldn't be an actual exception thrown here.

Reply:
done, thanks this helps!